### PR TITLE
Support all AWS Lambda runtimes including `nodejs10.x`

### PR DESCRIPTION
**Pull request by sebastienfi**
This adds support to all current AWS Lambda runtimes. 
Including newest `nodejs10.x`, `nodejs8.10` EOL, etc.
Instead of [dropping support to all but the latest](https://github.com/shelfio/aws-lambda-brotli-unpacker/issues/57#issuecomment-492956306), this makes it compatible with all.

This is a different approach to the problem from https://github.com/shelfio/aws-lambda-brotli-unpacker/pull/65 which makes it more generic - `iltorb@^2.4.3` is able to [detect the compilation runtime](https://github.com/MayhemYDG/iltorb/blob/master/README.md#install) and download the appropriate [lib version](https://github.com/MayhemYDG/iltorb/releases).

Also fix the JSDoc declaration of `unpack` function to keep linters happy.

Fixes https://github.com/shelfio/aws-lambda-brotli-unpacker/issues/57

### CODE REVIEW

**Comment by sebastienfi** [comment]
@samjeffress: can you review this within your context?

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "url": "shelf.io"
   },
   "engines": {
-    "node": ">=8.10"
+    "node": ">=10"
```
Comment (vladgolubev) [inline package.json]: Since this PR supports both node 8 & node 10, maybe leave `engines` setting as it was before?

**Comment by vladgolubev** [comment]
> which makes it more generic - iltorb@^2.4.3 is able to detect the compilation runtime  and download the appropriate lib version.

If I understand correctly, it will detect & download a proper binary version during npm package installation. Which means this will likely happen in some CI environment, which is far from the actual AWS Lambda environment. Correct me if I'm wrong

**Comment by maaraanas** [comment]
Old node versions, like 8 will be shut down in a month at the GCP for example. I think compatibility with older versions are not so big of a problem. LTS version is enough.

**Comment by AdamAH** [comment]
Will @shelf/aws-lambda-libreoffice be upgraded to use this latest version that supports Node 10 since being shut down in a the next month on AWS? Or will we have to manually update brotli-unpacker?

**Comment by vladgolubev** [comment]
Just to be clear, AWS won't shut down Node 8.10 runtime anytime soon, the nearest deadline is February 2020.

Meanwhile, looks like there is an awesome package [lambdafs](https://github.com/alixaxel/lambdafs) which does pretty much the same and supports both node 8 and node 10!

**Comment by AdamAH** [comment]
@vladgolubev Gotcha thanks. That's good to know its still a bit out. Regarding this package, since it's being used by @shelf/aws-lambda-libreoffice - without the need to fork it to point to lambdafs, this package will later on support Node 10 right? Confirming as I can wait on the upgrade our end now that I know we have a bit of time

**Comment by vladgolubev** [comment]
@AdamAH yes, we at Shelf depend on this package as well, so we're interested in updating it to support Node 10

**Comment by sebastienfi** [comment]
> > which makes it more generic - iltorb@^2.4.3 is able to detect the compilation runtime and download the appropriate lib version.
> 
> If I understand correctly, it will detect & download a proper binary version during npm package installation. Which means this will likely happen in some CI environment, which is far from the actual AWS Lambda environment. Correct me if I'm wrong

@vladgolubev It would be up to you to set up a CI that mimics AWS Lambda running environment. For example using Docker images made by [ImageCI](https://hub.docker.com/r/lambci/lambda/)

**Comment by sebastienfi** [comment]
@AdamAH @vladgolubev `@shelf/aws-lambda-libreoffice` does not seem to work yet on `nodejs10.x`. Some error about `"oosplash": error while loading shared libraries: libX11.so.6: cannot open shared object file: No such file or directory`. If you get there before I do, happy to review and test the code.

**Comment by samjeffress** [comment]
> @samjeffress: can you review this within your context?

I've had issues verifying changes to iltorb, as our primary use of this package is in @shelf/aws-lambda-libreoffice, and it's tests are throwing the `oosplash` error mentioned above. 

I think https://github.com/shelfio/aws-lambda-brotli-unpacker/pull/82 by @mrvisser adds a valuable test to verify the compression and provides some safety, and would be worth resolving.

**Comment by sebastienfi** [comment]
@samjeffress Thanks for the ping. My next window to work on this is the first week of December. I'll circle back in this thread.

**Comment by AdamAH** [comment]
hey guys, just wanted to check in, is there still a plan to support Node 10 or should people using this package move over to another package (before the jan lambda creation deadline and feb lambda update deadline)? Thanks!

**Comment by maaraanas** [comment]
@AdamAH yes, we will support node 10 till the end of the month. Already started moving forward. Stay in touch ;)

**Comment by AdamAH** [comment]
Awesome thanks @maaraanas :) Will be waiting for the update before we update our runtimes!

**Comment by maaraanas** [comment]
[https://github.com/shelfio/aws-lambda-brotli-unpacker/pull/83](83)

**Pull request closed without merging**